src/DeferredResult/DeferredResult.spec.ts

```diff
@@ -277,4 +277,73 @@ describe("DeferredResult", () => {
             }
         )
     })
+
+    describe("map", () => {
+        it("returns a Resolved with projected inner value if Resolved and Ok", () => {
+            const incr = (n: number) => n + 1
+            expect(
+                pipe(Deferred.resolved(Result.ok(22)), DeferredResult.map(incr))
+            ).toStrictEqual(Deferred.resolved(Result.ok(23)))
```

Comment:
❓ seeing this makes me wonder whether we could also easily add constructor functions for `DeferredResult.ok` and `DeferredResult.err`?

Reply:
https://github.com/fp-toolkit/fp-toolkit/pull/66/commits/56461bc3042a4c0f1f28ced77ad111edc3816758